src/native/libs/System.Security.Cryptography.Native.Apple/pal_keychain_ios.c

```diff
@@ -45,12 +45,10 @@ static int32_t EnumerateKeychain(CFStringRef matchType, CFArrayRef* pCertsOut)
         assert(result == NULL);
         status = noErr;
     }
-    else
+
+    if (result != NULL)
     {
-        if (result != NULL)
-        {
-            CFRelease(result);
-        }
+        CFRelease(result);
```

Comment:
This wrong, the array is returned to the managed code at the `*pCertsOut = (CFArrayRef)result;` line.

Reply:
Yeah, but it's retained before. That means it's doubly retained by this code path unless this `CFRelease` call is done unconditionally

Reply:
I see. Might be easier to skip the extra retain `CFRetain(result);` rather than doing retain followed by release.